overlord/snapstate/snapstate.go

```diff
@@ -2331,6 +2331,7 @@ func Switch(st *state.State, name string, opts *RevisionOptions) (*state.TaskSet
 		// set the from state (i.e. no change), they are overridden from opts as needed below
 		CohortKey: snapst.CohortKey,
 		Channel:   snapst.TrackingChannel,
+		Type:      snap.Type(snapst.SnapType),
```

Comment:
This change can be ignored, should not have been made as a part of this commit. Was undone in a later commit.